### PR TITLE
feature/#1734 Recreate the Core service

### LINKED ISSUE #1734: Recreate the Core service
After the Core service is replaced by the Orchestrator service, we want to keep the Core service compatible between `3.X` and `4.Y`.
The motivation is to give time to the developers to implement the changes and help them understand how to remain compatible. 

- Recreate the Core service class as a proxy to the orchestrator service class.
- Implement just a deprecation message saying the Core service does not exist anymore.

### PR DESCRIPTION
**Pull request by toan-quach**
#1734

### CODE REVIEW

File: taipy/core/_core.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+# specific language governing permissions and limitations under the License.
+
+from taipy.logger._taipy_logger import _TaipyLogger
+
+from .common._warnings import _warn_deprecated
+from .orchestrator import Orchestrator
+
+
+class Core:
+    """Deprecated. Use `Orchestrator` class with `tp.Orchestrator()` instead."""
```
Comment (jrobinAV) [inline taipy/core/_core.py]:
```suggestion
    """Deprecated. Use the `Orchestrator^` service class with `taipy.Orchestrator()` instead."""
```

File: taipy/core/_core.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+    __logger = _TaipyLogger._get_logger()
+
+    def __new__(cls) -> Orchestrator:  # type: ignore
+        _warn_deprecated("'Core'", suggest="the 'Orchestrator' class")
+        cls.__logger.warning(
+            "Class `Core` has been deprecated, an object of class `Orchestrator` will be created instead."
```
Comment (jrobinAV) [inline taipy/core/_core.py]:
```suggestion
            "The `Core` service is deprecated and replaced by the `Orchestrator` service. "
            "An `Orchestrator` instance has been instantiated instead."
```

**Review by jrobinAV — DISMISSED** [review]
It looks good to me.

File: tests/core/test_core.py [hunk 3]
```diff
@@ -12,105 +12,19 @@
...
-        with pytest.raises(ConfigurationUpdateBlocked):
-            Config.configure_data_node(id="i1")
-        orchestrator.stop()
-
-    @pytest.mark.standalone
-    def test_block_config_update_when_orchestrator_service_is_running_standalone_mode(self):
-        _OrchestratorFactory._dispatcher = None
+        assert isinstance(core, Orchestrator)
+        expected_message = (
+            "Class `Core` has been deprecated, an object of class `Orchestrator` will be created instead."
```
Comment (trgiangdo) [inline tests/core/test_core.py]: I believe we need to update this test message as well

**Review by trgiangdo — DISMISSED** [review]

File: tests/core/test_core.py (hunk 3 above)
Comment (toan-quach) [reply tests/core/test_core.py]: haha you're absolutely spotted on!! I just did and pushed the change also :D

**Review by trgiangdo — APPROVED** [review]
LGTM

**Pull request merged**